packages/interface-ipfs-core/src/files/ls.js

```diff
@@ -196,5 +198,14 @@ module.exports = (common, options) => {
       expect(files.length).to.equal(1)
       expect(files.filter(file => file.name === fileName)).to.be.ok()
     })
+
+    it('should respect timeout option when listing files', async () => {
+      await testTimeout(() => drain(ipfs.files.ls({
+        timeout: 1
+      })))
+
+      // ensures that the request that timed out has completed
+      await ipfs.files.stat('/')
```

Comment:
Hmm, I think it was necessary as `ipfsd-ctl` was failing to shut down nodes after the test run because the request was still in flight so datastores were open, etc but it shouldn't be - I've removed this everywhere.